R/FilteredData.R

```diff
@@ -452,7 +452,11 @@ FilteredData <- R6::R6Class( # nolint
     #'
     get_formatted_filter_state = function() {
       out <- c()
-      for (filtered_dataset in self$get_filtered_dataset()) out <- c(out, filtered_dataset$get_formatted_filter_state())
+      for (filtered_dataset in self$get_filtered_dataset()) {
+        if (length(filtered_dataset$get_filter_state()) != 0) {
+          out <- c(out, filtered_dataset$get_formatted_filter_state())
+        }
+      }
```

Comment:
Maybe this is more elegant than appending vector? Leaving up to you
```suggestion
      Filter(
        length
        sapply(
          self$get_filtered_dataset(), 
          function(filtered_dataset) {
            filtered_dataset$get_formatted_filter_state()
          }
        })
      )
```